capstone-backend/src/main/java/com/google/sps/data/BookList.java

```diff
@@ -17,77 +17,89 @@
  */
 public final class BookList {
 
-  private Long id;
-  private Long userID;
-  private String bookshelfID;
-  private Collection<Long> collaboratorsIDs;
+  private String id;
+  private String userID;
+  private Collection<String> gbookIDs;
+  private Collection<String> collaboratorsIDs;
 
-  public BookList(Long userID, String bookshelfID, Collection<Long> collaboratorsIDs, Long id) {
+  public BookList(String id, String userID, Collection<String> gbookIDs, Collection<String> collaboratorsIDs) {
+    this.id = id;
     this.userID = userID;
-    this.bookshelfID = bookshelfID;
+    this.gbookIDs = gbookIDs;
     this.collaboratorsIDs = collaboratorsIDs;
-    this.id = id;
   }
 
-  public BookList(Long userID, String bookshelfID, Collection<Long> collaboratorsIDs) {
-    this(userID, bookshelfID, collaboratorsIDs, null);
+  public BookList(String userID, Collection<String> gbookIDs, Collection<String> collaboratorsIDs) {
+    this("", userID, gbookIDs, collaboratorsIDs);
```

Comment:
I agree @steven-solar, not sure how we want to handle id creation. From the Firestore documentation (https://firebase.google.com/docs/firestore/manage-data/add-data) it says it will be overridden with the exact same id:

`If the document does not exist, it will be created. If the document does exist, its contents will be overwritten with the newly provided data, unless you specify that the data should be merged into the existing document`